solr/bin/solr

```diff
@@ -884,8 +887,12 @@ function stop_solr() {
     sleep 10
   fi
 
-  CHECK_PID=`ps auxww | awk '{print $2}' | grep -w $SOLR_PID | sort -r | tr -d ' '`
-  if [ "$CHECK_PID" != "" ]; then
+  STAT=`ps -o stat='' $SOLR_PID | tr -d ' '`
+  if [ "$STAT" == "Z" ]; then
+    # This can happen if, for example, you are running Solr inside a docker container with multiple processes
+    # rather than running it is as the only service. The --init flag on docker avoids that particular problem.
+    echo -e "Solr process $SOLR_PID has terminated abnormally. Solr has exited but a zombie process entry remains."
```

Comment:
Is this considered an error condition? I'm not sure. Maybe worthwhile to `exit 1` still? I'm not sure what your code calling this script would expect.

Reply:
Fair question. I think you're right that it should exit 1, since something is wrong.